### PR TITLE
CI: update all gha actions

### PR DESCRIPTION
**Pull request by andyfaff**

### CODE REVIEW

**Comment by andyfaff** [comment]
Updates made with `gha-update` package

File: .github/workflows/validate_refl1d.yml [hunk 1]
```diff
@@ -15,9 +15,9 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```
Comment (bmaranville) [inline .github/workflows/validate_refl1d.yml:18]: Why use a hash instead of a version tag?

**Comment by andyfaff** [comment]
The `gha-update` package does hashes by default. Using a script is nice because it does all actions automagically.
Also hashes are more secure. If the action author makes a point release then if you're using something like `v4` then you automatically start using that. If there's a security problem with that point release you're exposing the repo to that. If the repo permissions aren't locked down then that can create issues. By contrast hashes are fixed to a single version.
Using an updater script every so often seems like a good balance.

**Comment by bmaranville** [comment]
Good points... I try to stick mostly to actions provided directly by GitHub (where one might worry less about hijacked version tags) but it's also nice to offload some maintenance to gha-actions.

**Comment by bmaranville** [comment]
...though it seems like we're now trusting the `gha-update` package to not be malicious, instead

**Comment by andyfaff** [comment]
Sort of. You can check easily that youre not using a different action (unless there's a typosquat change). A single hash version is less likely to have a problem.

**Comment by andyfaff** [comment]
https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash

**Pull request merged**